src/app/shared/doc-viewer/doc-viewer.ts

```diff
@@ -38,13 +38,38 @@ export class DocViewer implements OnDestroy {
     }
   }
 
-  @Input() lines: readonly [number, number];
-
   @Output() contentRendered = new EventEmitter<HTMLElement>();
 
   /** The document text. It should not be HTML encoded. */
   textContent = '';
 
+  private static initExampleViewer(exampleViewerComponent: ExampleViewer,
+                                   example: string,
+                                   expanded: string | null,
+                                   file: string | null,
+                                   region: string | null) {
+    exampleViewerComponent.example = example;
+    if (expanded === 'true') {
+      exampleViewerComponent.view = 'full';
+    } else {
+      if (file) {
+        // if the html div has field `file` then it should be in compact view to show the code
+        // snippet
+        exampleViewerComponent.view = 'snippet';
+        exampleViewerComponent.showCompactToggle = true;
+        exampleViewerComponent.file = file;
+        if (region) {
+          // `lines` should only exist when `file` exists but not vice versa
+          // it is valid for embedded example snippets to show the whole file (esp short files)
+          exampleViewerComponent.region = region;
+        }
+      } else {
+        // otherwise it is an embedded demo
+        exampleViewerComponent.view = 'demo';
+      }
```

Comment:
Can we extract something in here to a helper method? It's visually hard to keep track of the `if...else` nesting.

Reply:
after removing `expanded` there is 1 less level, do you think that is sufficiently simple or should some parts still be extracted? I

Reply:
I think this is good now 👍